### PR TITLE
Add e2e test

### PR DESCRIPTION
**Pull request by avrittrohwer**
Fixes https://github.com/civiform/civiform/issues/3510.
Fixes https://github.com/civiform/civiform/issues/3974.

Tested on my personal fork: https://github.com/avrittrohwer/cloud-deploy-infra/actions/runs/3878728757/jobs/6615174557.

### CODE REVIEW

File: e2e-test/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# E2E Test
```
Comment (bion) [inline e2e-test/README.md:1]: suggestion (blocking): mention which cloud providers and templates are exercised with the current test setup and how to run the test manually

File: .github/workflows/e2e-test.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
+name: e2e-test
+on:
+  schedule:
+    - cron: '12 3 * * *'
+permissions:
+  id-token: write
+  contents: read
+jobs:
+  run-e2e:
```
Comment (bion) [inline .github/workflows/e2e-test.yaml:9]: suggestion (blocking): clarify that this is the AWS e2e test

File: e2e-test/civiform_config.sh [hunk 3]
```diff
@@ -0,0 +1,291 @@
+#! /usr/bin/env bash
```
Comment (bion) [inline e2e-test/civiform_config.sh:1]: suggestion (blocking): change filename to reflect that this tests the aws_oidc template

File: e2e-test/nuke.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+regions:
+  - us-east-1
+  - global
+
+account-blocklist:
+  - "664198874744"
```
Comment (bion) [inline e2e-test/nuke.yaml:6]: suggestion (blocking): add a comment for what this is

File: e2e-test/nuke.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+accounts:
+  "178882378957":
```
Comment (bion) [inline e2e-test/nuke.yaml:9]: suggestion (blocking): add a comment for what this is

File: e2e-test/run-nuke [hunk 6]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/expect
```
Comment (bion) [inline e2e-test/run-nuke:1]: suggestion (blocking): add TODO
```
TODO(https://github.com/civiform/civiform/issues/3974): remove use of expect.
```

**Review by bion — CHANGES_REQUESTED** [review]

File: .github/workflows/e2e-test.yaml [hunk 7]
```diff
@@ -0,0 +1,54 @@
+name: e2e-test
```
[the remark this answers is not in the document]
Comment (avrittrohwer) [reply .github/workflows/e2e-test.yaml:1]: Done.

File: .github/workflows/e2e-test.yaml [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+on:
+  schedule:
+    - cron: '12 3 * * *'
```
[the remark this answers is not in the document]
Comment (avrittrohwer) [reply .github/workflows/e2e-test.yaml:5]: Done.

File: e2e-test/README.md (hunk 1 above)
Comment (avrittrohwer) [reply e2e-test/README.md:1]: Added a coverage section.  There is not a good way to run the github action locally

File: .github/workflows/e2e-test.yaml (hunk 2 above)
Comment (avrittrohwer) [reply .github/workflows/e2e-test.yaml]: Changed to 'run-e2e-aws-oicd'.

File: e2e-test/civiform_config.sh (hunk 3 above)
Comment (avrittrohwer) [reply e2e-test/civiform_config.sh]: Changed to 'civiform_config_aws_oidc.sh'.

File: e2e-test/nuke.yaml (hunk 4 above)
Comment (avrittrohwer) [reply e2e-test/nuke.yaml]: Done.

File: e2e-test/nuke.yaml (hunk 5 above)
Comment (avrittrohwer) [reply e2e-test/nuke.yaml]: Done.

**Comment by avrittrohwer** [comment]
Blocked on https://github.com/civiform/civiform-deploy/pull/33/files

File: .github/workflows/e2e-test.yaml (hunk 7 above)
[the remark this answers is not in the document]
Comment (bion) [reply .github/workflows/e2e-test.yaml:1]: Please add newlines after lines 5, 8, 17, and 30

File: e2e-test/run-nuke (hunk 6 above)
Comment (bion) [reply e2e-test/run-nuke]: ah nice!

**Review by bion — APPROVED** [review]

**Comment by avrittrohwer** [comment]
Passing test in my fork: https://github.com/avrittrohwer/cloud-deploy-infra/actions/runs/3888225441/jobs/6635322434

**Comment by avrittrohwer** [comment]
Tested that triggering the workflow twice queues the second trigger: 
image

File: .github/workflows/e2e-test.yaml (hunk 7 above)
[the remark this answers is not in the document]
Comment (avrittrohwer) [reply .github/workflows/e2e-test.yaml:1]: Done

**Comment by nbeloglazov** [comment]
Woohoo-hoo!

**Pull request merged**